crates/sui-indexer-alt/src/handlers/sum_coin_balances.rs

```diff
@@ -59,38 +60,49 @@ impl Processor for SumCoinBalances {
                 }
             }
 
-            // Deleted and wrapped coins
+            // Do a fist pass to add updates without their associated contents into the `values`
+            // mapping, based on the transaction's object changes.
             for change in tx.effects.object_changes() {
```

Comment:
`TransactionEffectsAPI` has methods of 
```
    fn created(&self) -> Vec<(ObjectRef, Owner)>;
    fn mutated(&self) -> Vec<(ObjectRef, Owner)>;
    fn unwrapped(&self) -> Vec<(ObjectRef, Owner)>;
    fn deleted(&self) -> Vec<ObjectRef>;
    fn unwrapped_then_deleted(&self) -> Vec<ObjectRef>;
    fn wrapped(&self) -> Vec<ObjectRef>;
```
it seems cleaner to use that instead of classifying object changes again?

Reply:
Maybe it's just me, but I don't find that cleaner. For me, `object_changes` is more uniform, because it separates out the detail of whether the object's ID was created/deleted from whether the object was added or removed from the global store, which means you can focus on just one part.

For example, in this case, we only care about visibility in the global store, and we are able to use the presence/absence of input/output versions to enumerate every possible case. As far as I can tell, if I wanted to do the same thing with the EffectsV1 compatibility APIs:

- I would need to merge together the cases that I cared about (`unwrapped` and `created` go together, `wrapped` and `deleted` go together, `mutated` goes on its own),
- I don't get the compiler's help to tell me I've handled every case,
- the code around the part that figures the update kind would need to be duplicated, and
- when these compatibility APIs get called on V2 effects (the majority on chain these days), they will scan overall object changes every time, filtering changes into these buckets, only for us to merge those buckets back together.

Maybe I'm missing something though -- @gegaowp, if you see a clearer path, can you elaborate?